src/main/java/com/growup/pms/task/repository/TaskQueryRepositoryImpl.java

```diff
@@ -22,41 +28,51 @@ public class TaskQueryRepositoryImpl implements TaskQueryRepository {
     private final JPAQueryFactory queryFactory;
 
     @Override
-    public List<TaskResponse> getTasksByStatusId(Long statusId) {
+    public Map<Long, List<TaskResponse>> getTasksByProjectId(Long projectId) {
         List<Long> ids = queryFactory
                 .select(task.id)
                 .from(task)
-                .leftJoin(task.status, status)
                 .leftJoin(task.user, user)
+                .join(task.status, status)
+                .join(task.status.project, project)
                 .where(
-                        isStatusId(statusId)
+                        projectId != null ? task.status.project.id.eq(projectId) : null
                 )
                 .orderBy(task.sortOrder.asc())
                 .fetch();
 
-        if (ids == null || ids.isEmpty()) {
-            return Collections.emptyList();
+        if (ids.isEmpty()) {
+            return Collections.emptyMap();
         }
 
-        return queryFactory
-                .select(Projections.constructor(TaskResponse.class,
+        List<Tuple> results = queryFactory
+                .select(task.status.id, Projections.constructor(TaskResponse.class,
                         task.id,
                         task.status.id,
                         task.name,
-                        task.user.username,
+                        task.user != null ? task.user.username : null,
                         task.sortOrder
                 ))
                 .from(task)
-                .leftJoin(task.status, status)
                 .leftJoin(task.user, user)
+                .join(task.status, status)
+                .join(task.status.project, project)
                 .where(
                         task.id.in(ids)
                 )
                 .orderBy(task.sortOrder.asc())
                 .fetch();
-    }
 
-    private BooleanExpression isStatusId(Long statusId) {
-        return statusId != null ? task.status.id.eq(statusId) : task.status.isNull();
+        return results.stream()
+                .sorted(Comparator.comparingInt(t -> t.get(1, TaskResponse.class).sortOrder()))
+                .collect(
+                        groupingBy(
+                                tuple -> tuple.get(1, TaskResponse.class).statusId(),
+                                Collectors.mapping(
+                                        tuple -> tuple.get(1, TaskResponse.class),
+                                        toList()
+                                )
+                        )
+                );
```

Comment:
매핑 로직을 이전 연산에서 수행하면 이렇게 간소화해볼 수 있을 것 같은데 제대로 동작할지 확신이 없네요 :sweat_smile: 

```suggestion
        return results.stream()
                .map(tuple -> tuple.get(1, TaskResponse.class))
                .collect(groupingBy(task -> Objects.requireNonNull(task).statusId(), toList()));
```